Userland/Libraries/LibWeb/CSS/CSSPropertyRule.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024, Alex Ungurianu <alex@ungurianu.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/CSSPropertyRulePrototype.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/CSS/CSSPropertyRule.h>
+#include <LibWeb/CSS/Parser/Parser.h>
+
+namespace Web::CSS {
+
+JS_DEFINE_ALLOCATOR(CSSPropertyRule);
+
+JS::NonnullGCPtr<CSSPropertyRule> CSSPropertyRule::create(JS::Realm& realm, FlyString name, FlyString syntax, bool inherits, Optional<String> initial_value)
+{
+    return realm.heap().allocate<CSSPropertyRule>(realm, realm, move(name), move(syntax), inherits, move(initial_value));
+}
+
+CSSPropertyRule::CSSPropertyRule(JS::Realm& realm, FlyString name, FlyString syntax, bool inherits, Optional<String> initial_value)
+    : CSSRule(realm)
+    , m_name(move(name))
+    , m_syntax(move(syntax))
+    , m_inherits(inherits)
+    , m_initial_value(move(initial_value))
+{
+}
+
+void CSSPropertyRule::initialize(JS::Realm& realm)
+{
+    Base::initialize(realm);
+    WEB_SET_PROTOTYPE_FOR_INTERFACE(CSSPropertyRule);
+}
+
+// https://www.w3.org/TR/cssom-1/#serialize-a-css-rule
+String CSSPropertyRule::serialized() const
+{
+    // Note: The spec doesn't cover this yet. Using the WPT `cssText` tests instead for inspiration
+    // Should be pretty close!
+
+    StringBuilder builder;
+
+    builder.appendff("@property {}", name());
```

Comment:
The formatter here doesn't do exactly what WPT expects. It fails one of the test cases due to how a tab character is encoded:

```
assert_equals: expected "@property --tab\\9 tab { syntax: \"*\"; inherits: true; }" but got "@property --tab\ttab { syntax: \"*\"; inherits: true; }"
```

Reply:
I think you want to use `serialize_an_identifier()` from the `CSS/Serialize.h` header.

Also, I managed to find a spec for serializing this: https://drafts.css-houdini.org/css-properties-values-api/#serialize-a-csspropertyrule It wasn't obvious to find. It mentions using the `serialize foo` algorithms that are in that header.

Reply:
That worked. Fixed